### PR TITLE
Ava + Grunt - Multi level parallel test runner

### PR DESCRIPTION
**Pull request by subash-canapathy**
Adding a new Recipe:
Ava + Grunt - Multi level parallel test harness for end to end tests

### CODE REVIEW

**Comment by jamestalmage** [comment]
A few notes:

- I don't really understand what this recipe accomplishes, or how it's worthwhile for our users. Exactly what problem does it solve? That should probably be made clear at the top of the document (I advise you to convince us of the need for this recipe before investing more time in it though).

- I think some of the language needs to be reworked. It feels a little too flowery / sales pitchy at some points:
  -  `note how much time the tests take ...`  - It's unrealistic to expect them to actually run the code from the recipe. Show a screen shot. Or just dryly stating `this way is slower because ...` is fine to.
  - `! #mindblowing` - I'm all for exciting language and exclamation points where warranted, but let's do it while communicating something of value. This doesn't add value, a statement like `That's a whopping 300% speedup!` communicates facts while still slipping in some fun language that keeps it a light read.

All that said, your first task really is convincing us of the problem this solves, and then that this is the best way to solve it.

File: docs/recipes/grunt-ava-parallel-tests.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+* when we get a new driver (with nanotime based ID)
+* when a test uses the driver
+* when the driver gets cleaned up.
+
+## Organizing test suites
+
+The additional parallel dimension here which is not provided by Ava comes from Grunt. The example uses a grunt plugin (from the developer who wrote Ava) `grunt-concurrent`. Its goal is to take a list of grunt tasks and be able to run them concurrently.
+To make sure we can set the environment vars and other configuration options per suite/combination we leverage `grunt-env` and some task compositions to make it work.
+
+Ava test suites are composed in `grunt-shell` blocks (until we have cli options support on `grunt-ava`)
```
Comment (sotojuan) [inline docs/recipes/grunt-ava-parallel-tests.md]: Nitpick: It's "AVA" in all caps.

File: docs/recipes/grunt-ava-parallel-tests.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+Multi level parallel tests runs where the time taken is only in the order of your longest running test !
+
+##Sample Test harness
+https://github.com/subash-canapathy/grunt-ava-parallel-tests
+
+You can use this as a bootstrap test runner / harness to build automated test framework on. It is designed with webdriver/appium end to end tests in mind so that you can run your UI tests with only your grid as the limiting factor.
+## Usage
```
Comment (sotojuan) [inline docs/recipes/grunt-ava-parallel-tests.md]: You should have a blank like before and after Markdown headings (you do elsewhere, so this is more about consistency).

**Comment by sindresorhus** [comment]
Closing for lack of response.

**Pull request closed without merging**